llvm/test/CodeGen/LoongArch/lsx/vector-fp-conv.ll

```diff
@@ -0,0 +1,123 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc --mtriple=loongarch64 --mattr=+lsx < %s | FileCheck %s
+
+;; ceilf
+define float @ceil_f32(float %i) nounwind {
+; CHECK-LABEL: ceil_f32:
```

Comment:
what did we generate before this commit? `Expand`?

Reply:
Thanks for your review!
You are right. Without this commit, `llvm.{ceil/floor/trunc/roundeven}` intrinsics were expanded at the loongarch backend. A call to the function in libm was generated.

Reply:
This commit works only when lsx is enabled because of the lack of explicit rounding mode scalar fp instructions. So I just update this commit (add tests without lsx). The LSX results before this commit were all same as NOLSX.

I didn't measure the cost for clang. We can see the result for gcc in https://gcc.gnu.org/pipermail/gcc-cvs/2023-November/394218.html : `much faster than calling a libc function: on LA464 and LA664 there is a 3x ~ 5x speed up`.

Reply:
Thanks for your link, very solid.